src/pages/library/datasetBuilder/CohortEditor.test.ts

```diff
@@ -0,0 +1,218 @@
+import { render, screen } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+import _ from 'lodash/fp';
+import { h } from 'react-hyperscript-helpers';
+import { dummyDatasetDetails } from 'src/libs/ajax/DatasetBuilder';
+import {
+  CohortEditor,
+  createCriteriaFromType,
+  createCriteriaViewComponent,
+  CriteriaGroupView,
+} from 'src/pages/library/datasetBuilder/CohortEditor';
+import {
+  Cohort,
+  CriteriaGroup,
+  DomainCriteria,
+  newCohort,
+  newCriteriaGroup,
+  ProgramDataListCriteria,
+  ProgramDataRangeCriteria,
+} from 'src/pages/library/datasetBuilder/dataset-builder-types';
+import { HomepageState } from 'src/pages/library/datasetBuilder/DatasetBuilder';
+import { datasetBuilderCohorts } from 'src/pages/library/datasetBuilder/state';
+
+describe('CohortEditor', () => {
+  const datasetDetails = dummyDatasetDetails;
+
+  it('renders unknown criteria', () => {
+    const criteria = { name: 'bogus', invalid: 'property' };
+
+    // The 'as any' is required to create an invalid criteria for testing purposes.
+    render(createCriteriaViewComponent(_.noop)(criteria as any));
+
+    expect(screen.queryByText(criteria.name)).toBeFalsy();
+    expect(screen.queryByText('Unknown criteria type')).toBeTruthy();
+  });
+
+  it('renders domain criteria', () => {
+    const criteria = createCriteriaFromType({ id: 0, category: 'category', values: ['value'] }) as DomainCriteria;
```

Comment:
I see a lot of "as TypeX" casting in these tests.  These should all be moved to stronger type safety with:

const criteria: DomainCriteria = createBlahBlah();